src/multivariate/dirichlet.jl

```diff
@@ -154,20 +154,49 @@ end
 
 # sampling
 
+function _rand_handle_overflow!(
+    rng::AbstractRNG,
+    d::Union{Dirichlet,DirichletCanon},
+    x::AbstractVector{<:Real}
+    )
+    Σ = sum(x)
+    if Σ == 0.0
+        # Distribution behavior approaches categorical as Σα -> 0
+        α = d.alpha
+        iΣα = inv(d.alpha0)
+        if isinf(iΣα)
+            # Dirichlet with ALL deeply subnormal parameters
+            α .*= floatmax(eltype(α))
```

Comment:
Though this new function is only being used inside of mutating functions, those mutating functions only mutated `x`, whereas this is modifying the `Distribution`'s parameter array. I would find that behavior unexpected as a user.